gio/src/app_info.rs

```diff
@@ -93,19 +93,14 @@ impl<O: IsA<AppInfo>> AppInfoExtManual for O {
     #[cfg_attr(feature = "dox", doc(cfg(feature = "v2_60")))]
     fn launch_uris_async_future<P: IsA<AppLaunchContext> + Clone + 'static>(
         &self,
-        uris: &[&str],
+        uris: &[impl AsRef<str>],
         context: Option<&P>,
     ) -> Pin<Box_<dyn std::future::Future<Output = Result<(), glib::Error>> + 'static>> {
-        let uris = uris.iter().copied().map(String::from).collect::<Vec<_>>();
         let context = context.map(ToOwned::to_owned);
         Box_::pin(crate::GioFuture::new(self, move |obj, send| {
             let cancellable = Cancellable::new();
-            let uris = uris
-                .iter()
-                .map(::std::borrow::Borrow::borrow)
-                .collect::<Vec<_>>();
             obj.launch_uris_async(
-                uris.as_ref(),
+                uris,
```

Comment:
Not sure this will actually compile, maybe after the trait bounds validation the compiler might complain about this. But I can work out a solution for that later, that is fine..